coding-challenges.js

```diff
@@ -1,60 +1,118 @@
 
 
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
-
-
-
-
-
-var cubeAndSum1 = [2, 3, 4]
+// // --------------------
+// 2) Create a function that 
+// takes in an array of numbers and 
+// returns the sum of all the numbers cubed.
+var cubeAndSum1 = [2, 3, 4];
 // Expected output: 99
-var cubeAndSum2 = [0, 5, 10]
+var cubeAndSum2 = [0, 5, 10];
 // Expected output: 1125
 
+const arrCube = (cube) => {
+    let newCubeArray = 0;
+    for (let i=0; i< cube.length; i++) {
+        newCubeArray = (cube[0]**3+cube[1]**3+cube[2]**3)
+    }
+    return newCubeArray;
+};
 
-
-// --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
+console.log(arrCube(cubeAndSum1));   
+console.log(arrCube(cubeAndSum2));
 
 
-
-
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
+// --------------------
+// 3) Create a function that 
+// takes an array of numbers and 
+// returns an array of the minimum and maximum numbers in that order.
+var nums1 = [3, 56, 90, -8, 0, 23, 6];
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, -59, 8, 24]
+var nums2 = [109, 5, 9, -59, 8, 24];
 // Expected output: [-59, 109]
 
+function arrMinMax(nummm) {
+  return [Math.min(...nummm), Math.max(...nummm)];
+}
 
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
+console.log(arrMinMax(nums1));   
+console.log(arrMinMax(nums2));
 
 
+// --------------------
+// 4) Create a function that 
+// takes in a string and 
+// returns a string with every other letter capitalized.
 
 
-
-var testString1 = "albatross"
+var testString1 = "albatross";
 // Expected output: "aLbAtRoSs"
-var testString2 = "jabberwocky"
+var testString2 = "jabberwocky";
 // Expected output: "jAbBeRwOcKy"
 
+function strCapIt (str) {
+    let strSuper = '';
+    for ( i = 0; i < str.length; i++) {
+        strSuper += i % 2 == 1 ? str.charAt(i).toUpperCase() : str.charAt(i);
+  }
+  return strSuper;  
+}
 
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
-
+console.log(strCapIt(testString1));   
+console.log(strCapIt(testString2));
 
 
 
+// --------------------
+// 5) Create a function that 
+// takes in two arrays as arguments and 
+// returns one array with no duplicate values. 
+// STRETCH: Use the spread operator.
 
-var arr1 = [3, 7, 10, 5, 4, 3, 3]
-var arr2 = [7, 8, 2, 3, 1, 5, 4]
+var arr1 = [3, 7, 10, 5, 4, 3, 3];
+var arr2 = [7, 8, 2, 3, 1, 5, 4];
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+var arr3 = [...arr1, ...arr2];
+
+// arr3.sort();
+
+function duplicateArray (removeDouble)  {
+    var a = [];
+    for (var i=0, l=removeDouble.length; i<l; i++)
+        if (a.indexOf(removeDouble[i]) === -1)
+            a.push(removeDouble[i]);
+    return a;
+}
```

Comment:
great! how can we move `arr3` _into_ the function in order for it to be as re-usable as possible?